modules/atlas-basic/variables.tf

```diff
@@ -0,0 +1,170 @@
+  type        = string
+}
+variable "private_key" {
+  description = "Private API key to authenticate to Atlas"
+  type        = string
+}
+
+# project
+variable "project_name" {
+  description = "Atlas project name"
+  default     = "TenantUpgradeTest"
+  type        = string
+}
+
+#cluster
+variable "cluster_name" {
+  description = "Atlas cluster name"
+  default     = "cluster"
+  type        = string
+}
+
+variable "cluster_type" {
+  description = "Atlas cluster type"
+  default     = "REPLICASET"
+  type        = string
+}
+
+variable "num_shards" {
+    description = "Atlas cluster number of shards"
+    default     = 1
+    type        = number
+}
+
+variable "priority" {
+    description = "Atlas cluster priority"
+    default     = 7
+    type        = number
+}
+
+variable "read_only_nodes" {
+    description = "Atlas cluster number of read only nodes"
+    default     = 0
+    type        = number
+}
+variable "electable_nodes" {
+    description = "Atlas cluster number of electable nodes"
+    default     = 3
+    type        = number
+}
+
+variable "auto_scaling_disk_gb_enabled" {
+    description = "Atlas cluster auto scaling disk enabled"
+    default     = false
+    type        = bool
+}
+
+variable "disk_size_gb" {
+    description = "Atlas cluster disk size in GB"
+    default     = 10
+    type        = number
+}
+variable "provider_name" {
+  description = "Atlas cluster provider name"
+  default     = "AWS"
+  type        = string
+}
+variable "backing_provider_name" {
+  description = "Atlas cluster backing provider name"
+  default = "AWS"
+  type        = string
+}
+variable "provider_instance_size_name" {
+  description = "Atlas cluster provider instance name"
+  default     = "M10"
+  type        = string
+}
+
+variable "region" {
+    description = "Atlas cluster region"
+    default     = "US_EAST_1"
+    type        = string
+}
+variable "aws_region"{
+    description = "AWS region"
+    default     = "us-east-1"
+    type        = string
+}
+
+variable "mongo_version" {
+  description = "Atlas cluster version"
+  default     = "4.4"
+  type        = string
+}
+
+
+variable "user" {
+  description = "MongoDB Atlas User"
+  type        = list(string)
+  default     = ["dbuser1", "dbuser2"]
+}
+variable "password" {
+  description = "MongoDB Atlas User Password"
+  type        = list(string)
+}
+variable "database_name" {
+  description = "The Database in the cluster"
+  type        = list(string)
+}
+
+# database user
+variable "role_name" {
+    description = "Atlas database user role name"
+    default     = "readWrite"
+    type        = string
+}
+
+# IP Access List
+variable "cidr_block" {
+    description = "IP Access List CIDRs"
+    type        = list(string)
+    default     = ["10.1.0.0/16","12.2.0.0/16"]
+}
+
+variable "ip_address" {
+    description = "IP Access List IP Addresses"
+    type        = list(string)
+    default     = ["208.169.90.207","0.0.0.0"]
```

Comment:
we should not provide default here as we don't now the IP addressed of the machines that will access the database
```suggestion
```

Reply:
Reverted